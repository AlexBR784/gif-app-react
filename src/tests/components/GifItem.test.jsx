import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("GifItem tests", () => {
  const title = "titulo",
    url = "http://example.com/";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  
  test("Debe de mostrar la imagen con la url indicada", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByRole("img").src).toBe(url);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(title).toBeTruthy();
  });
  
});
