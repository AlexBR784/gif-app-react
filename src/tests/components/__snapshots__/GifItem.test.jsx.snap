// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="http://example.com/"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
